sidechain/src/tests.rs

```diff
@@ -26,259 +26,220 @@ fn get_signer(pubkey: &[u8; 32]) -> AccountId {
-		let header2 = new_header(2, header1.hash());
-		let header3 = new_header(3, header2.hash());
-		let header4 = new_header(4, header3.hash());
-		let header5 = new_header(5, header4.hash());
-
-		assert_ok!(confirm_block(header1, true));
-		assert_eq!(Sidechain::latest_sidechain_header(shard7).block_number, 1);
-		assert_ok!(confirm_block(header2, true));
-		assert_eq!(Sidechain::latest_sidechain_header(shard7).block_number, 2);
-		assert_ok!(confirm_block(header3, true));
-		assert_eq!(Sidechain::latest_sidechain_header(shard7).block_number, 3);
-		assert_ok!(confirm_block(header4, true));
-		assert_eq!(Sidechain::latest_sidechain_header(shard7).block_number, 4);
-		assert_ok!(confirm_block(header5, true));
-		assert_eq!(Sidechain::latest_sidechain_header(shard7).block_number, 5);
+		register_enclave7();
+
+		assert_ok!(confirm_block7(1, H256::random(), true));
+		assert_eq!(Sidechain::latest_sidechain_block_confirmation(shard7).block_number, 1);
+		assert_ok!(confirm_block7(2, H256::random(), true));
+		assert_eq!(Sidechain::latest_sidechain_block_confirmation(shard7).block_number, 2);
+		assert_ok!(confirm_block7(3, H256::random(), true));
+		assert_eq!(Sidechain::latest_sidechain_block_confirmation(shard7).block_number, 3);
+		assert_ok!(confirm_block7(4, H256::random(), true));
+		assert_eq!(Sidechain::latest_sidechain_block_confirmation(shard7).block_number, 4);
+		assert_ok!(confirm_block7(5, H256::random(), true));
+		assert_eq!(Sidechain::latest_sidechain_block_confirmation(shard7).block_number, 5);
 	})
 }
 
 #[test]
-fn confirm_proposed_sidechain_first_proposed_block() {
+fn confirm_imported_sidechain_first_imported_block() {
 	new_test_ext().execute_with(|| {
 		Timestamp::set_timestamp(TEST7_TIMESTAMP);
 		let shard7 = H256::from_slice(&TEST7_MRENCLAVE);
 
-		register_enclave();
-
-		let header1 = new_header(1, H256::default());
-		let header2 = new_header(2, header1.hash());
-		let mut header3a = new_header(3, header2.hash());
-		header3a.block_data_hash = [2; 32].into();
-		let mut header3b = new_header(3, header2.hash());
-		header3b.block_data_hash = [3; 32].into();
-
-		assert_ok!(confirm_block(header1, true));
-		assert_eq!(Sidechain::latest_sidechain_header(shard7).block_number, 1);
-		assert_ok!(confirm_block(header3a, false));
-		assert_eq!(Sidechain::latest_sidechain_header(shard7).block_number, 1);
-		assert_ok!(confirm_block(header3b, false));
-		assert_eq!(Sidechain::latest_sidechain_header(shard7).block_number, 1);
-		assert_ok!(confirm_block(header2, true));
-		assert_eq!(Sidechain::latest_sidechain_header(shard7).block_number, 3);
-		assert_eq!(
-			Sidechain::latest_sidechain_header(shard7).block_data_hash,
-			header3a.block_data_hash
-		);
+		register_enclave7();
+
+		let hash = H256::random();
+
+		assert_ok!(confirm_block7(1, H256::random(), true));
+		assert_eq!(Sidechain::latest_sidechain_block_confirmation(shard7).block_number, 1);
+		assert_ok!(confirm_block7(3, hash, false));
```

Comment:
I know this hasn't been here before, but it was a little pain to understand this test. Can you add a comment here like
`// Queue block number 3. Should not be imported right now, because confirmation of block number 2 is missing`